### PR TITLE
zydis: init at 3.1.0

### PR DESCRIPTION
**Pull request by jbcrail**
###### Motivation for this change

Add [zydis](https://github.com/zyantific/zydis).

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -10484,4 +10484,10 @@
     github = "zupo";
     githubId = 311580;
   };
+  jbcrail = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:10487]: Please put this into a separate commit with the message ``maintainers: add jbcrail``.

File: pkgs/development/libraries/zydis/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    sha256 = "0x2lpc33ynd0zzirdxp2lycvg3545wh1ssgy4qlv81471iwwzv6b";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  meta = with lib; {
+    description = "Fast and lightweight x86/x86-64 disassembler library";
+    license = licenses.mit;
+    maintainers = [ stdenv.lib.maintainers.jbcrail ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/zydis/default.nix:20]:
```suggestion
    maintainers = [ maintainers.jbcrail ];
```

File: pkgs/development/libraries/zydis/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    platforms = with platforms; [ "i686-linux" "x86_64-linux" "x86_64-darwin" ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/zydis/default.nix:21]:
```suggestion
    platforms = platforms.all;
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -29702,4 +29702,6 @@ in
   lc3tools = callPackage ../development/tools/lc3tools {};
 
   zktree = callPackage ../applications/misc/zktree {};
+
+  zydis = callPackage ../development/libraries/zydis {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:29706]: Please add it into a fitting category.

```suggestion
  zydis = callPackage ../development/libraries/zydis { };
```

File: pkgs/development/libraries/zydis/default.nix [hunk 5]
```diff
@@ -0,0 +1,23 @@
+{ lib, stdenv, fetchFromGitHub, cmake }:
+
+stdenv.mkDerivation rec {
+  pname = "zydis";
+  version = "3.1.0";
+
+  src = fetchFromGitHub {
+    repo = "zydis";
+    owner = "zyantific";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/zydis/default.nix:9]:
```suggestion´
    owner = "zyantific";
    repo = "zydis";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 4 above)
Comment (jbcrail) [reply pkgs/top-level/all-packages.nix]: Thanks, I moved this to `development/tools` instead.

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -29702,4 +29702,6 @@ in
   lc3tools = callPackage ../development/tools/lc3tools {};
 
   zktree = callPackage ../applications/misc/zktree {};
+
+  zydis = callPackage ../development/tools/zydis { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:29706]:
```suggestion
  zydis = callPackage ../development/libraries/zydis { };
```

Comment (jbcrail) [reply pkgs/top-level/all-packages.nix]: Thanks, I reverted my previous change.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110642` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>zydis</li>
</ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -29717,4 +29717,6 @@ in
   lc3tools = callPackage ../development/tools/lc3tools {};
 
   zktree = callPackage ../applications/misc/zktree {};
+
+  zydis = callPackage ../development/libraries/zydis { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:29721]: Please sort this into a fitting category and sort there alphabetically.

Comment (jbcrail) [reply pkgs/top-level/all-packages.nix]: Oops, I just realized this file is grouped by categories. I made the requested change, thanks.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110642` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>zydis</li>
</ul>
</details>

File: maintainers/maintainer-list.nix [hunk 8]
```diff
@@ -10496,4 +10496,10 @@
...
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:10499]: I missed that you did not sort this. Going to fix it in a PR.

Comment (jbcrail) [reply maintainers/maintainer-list.nix:10499]: My apologies, thanks.
